doc/source/whatsnew/v1.3.0.rst

```diff
@@ -682,6 +682,7 @@ Deprecations
 - Deprecated passing arguments as positional in :meth:`DataFrame.clip` and :meth:`Series.clip` (other than ``"upper"`` and ``"lower"``) (:issue:`41485`)
 - Deprecated special treatment of lists with first element a Categorical in the :class:`DataFrame` constructor; pass as ``pd.DataFrame({col: categorical, ...})`` instead (:issue:`38845`)
 - Deprecated passing arguments as positional (except for ``"method"``) in :meth:`DataFrame.interpolate` and :meth:`Series.interpolate` (:issue:`41485`)
+- Deprecated the :attr:`Timestamp.freq` attribute.  For the properties that use it (``is_month_start``, ``is_month_end``, ``is_quarter_start``, ``is_quarter_end``, ``is_year_start``, ``is_year_end``), when you have a ``freq``, use e.g. ``freq.is_month_start(ts)`` (:issue:`15146`)
```

Comment:
I don't find the explanation of "for properties that use it .., when you have a `freq`, ..." very clear. Adding an actual example might clarify.

First, I assume people who do `ts.is_month_start` (with `ts` being a Timestamp) don't necessarily know that they are relying on some `freq`. 
Also, the attributes are not going away, right? It's only the fastpath for computing it based on the freq that goes away? Are there actually cases that will change behaviour, or is it only performance? (and if only performance, that might not matter much for scalars, or maybe not worth warning about).

Reply:
> Are there actually cases that will change behaviour, or is it only performance?

Yes, e.g.

```
dti = pd.date_range("2017-01-01", periods=30, freq="B")
dti[0].is_month_start  # <- True now, False once freq is removed
```

Reply:
And in principle we can know when this will change? (depending on the `freq`, we can know whether it can give different answers?) 
Could we warn only in those cases?

Reply:
Good idea, will do

Reply:
Updated to warn in fewer cases.  Still haven't updated the whatsnew note per the first comment here.  suggestions on wording?

Reply:
i agree here. i think we are simply deprecating freq. why do we need to even say anything about `is_month_start` for example?

Reply:
bc the result of is_month_start is going to change in some cases

Reply:
One consequence of deprecating Timestamp.freq is that when having a timeseries with a freq, the following equivalency won't no longer always hold:

```
dtidx.is_month_start[0] == dtidx[0].is_month_start
```
That's a bit unfortunate, I would say, and probably something we should explicitly document.

Reply:
ok for this as a followup documentation (but for 1.3)